### PR TITLE
Refactor file_paths tuple into ModuleSpec (preliminary)

### PR DESCRIPTION
**Pull request by Northbadge**
Preliminary patch to refactoring file_paths tuple usage completely.

Allows distributed thinlto corpora to be optionally trained on independent of runner type

Prereq: #55 #56

### CODE REVIEW

**Comment by Northbadge** [comment]
Good for review now, just rebased it to `main`

File: compiler_opt/rl/corpus.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from dataclasses import dataclass
+from typing import Tuple
+
+
+@dataclass(frozen=True)
+class ModuleSpec:
+  """Dataclass describing an input module and its compilation command options.
+  """
+  name: str
+  has_thinlto: bool = False
```
Comment (mtrofin) [inline compiler_opt/rl/corpus.py:26]: nit: is_thinlto

**Comment by mtrofin** [comment]
lgtm, just a nit

File: compiler_opt/rl/local_data_collector.py [hunk 2]
```diff
@@ -76,15 +77,13 @@ def _join_pending_jobs(self):
     logging.info('Waiting for pending work from last iteration took %f',
                  time.time() - t1)
 
-  def _schedule_jobs(
-      self, policy_path, sampled_file_paths
-  ) -> List[worker.WorkerFuture[compilation_runner.CompilationResult]]:
```
Comment (yundiqian) [inline compiler_opt/rl/local_data_collector.py:81]: accidental deletion?

Comment (mtrofin) [reply compiler_opt/rl/local_data_collector.py:81]: looks that way - I'll fix it (thanks for finding it!)

**Pull request merged**